gen/src/main/java/com/softlayer/api/gen/ClassWriter.java

```diff
@@ -377,8 +377,14 @@ public ClassWriter emitType() throws IOException {
         
         // Now the service
         if (!type.meta.noservice) {
-            if (type.meta.properties.containsKey("id")) {
-                if (type.meta.properties.containsKey("globalIdentifier")) {
+
+            Boolean containsId = type.meta.properties.containsKey("id")
+                    || type.baseMeta.properties.containsKey("id");
```

Comment:
Keeping a reference to the `Meta` object in `ClassWriter` is better than not supporting recursion IMO.